Lib/fontbakery/specifications/ufo_sources.py

```diff
@@ -0,0 +1,106 @@
+# -*- coding: utf-8 -*-
+#
+# This file has been automatically formatted with `yapf --style '
+# {based_on_style: google}'` and `docformatter`.
+from __future__ import (absolute_import, print_function, unicode_literals,
+                        division)
+
+from fontbakery.checkrunner import (INFO, WARN, ERROR, SKIP, PASS, FAIL,
+                                    Section, Spec)
+import os
+from fontbakery.callable import condition, check
+from fontbakery.message import Message
+from fontbakery.constants import (
+    # TODO: priority levels are not yet part of the new runner/reporters.
+    # How did we ever use this information?
+    # Check priority levels:
+    CRITICAL,
+    IMPORTANT
+    #     , NORMAL
+    #     , LOW
+    #     , TRIVIAL
+)
+
+default_section = Section('Default')
+
+
+class UFOSpec(Spec):
+
+    def setup_argparse(self, argument_parser):
+        """Set up custom arguments needed for this spec."""
+        import glob
+        import logging
+        import argparse
+
+        def get_fonts(pattern):
+
+            fonts_to_check = []
+            # use glob.glob to accept *.ufo
+
+            for fullpath in glob.glob(pattern):
+                if fullpath.endswith(".ufo"):
+                    fonts_to_check.append(fullpath)
+                else:
+                    logging.warning(
+                        ("Skipping '{}' as it does not seem "
+                         "to be valid UFO source directory.").format(fullpath))
+            return fonts_to_check
+
+        class MergeAction(argparse.Action):
+
+            def __call__(self, parser, namespace, values, option_string=None):
+                target = [item for l in values for item in l]
+                setattr(namespace, self.dest, target)
+
+        argument_parser.add_argument(
+            'fonts',
+            nargs='+',
+            type=get_fonts,
+            action=MergeAction,
+            help='font file path(s) to check.'
+            ' Wildcards like *.ufo are allowed.')
+        return ('fonts',)
+
+
+# ----------------------------------------------------------------------------
+# This variable serves as an exportable anchor point, see e.g. the
+# Lib/fontbakery/commands/check_ufo_sources.py script.
+specification = UFOSpec(
+    default_section=default_section,
+    iterargs={'font': 'fonts'},
+    derived_iterables={'ufo_sources': ('ufo_source', True)}
+)
+
+register_check = specification.register_check
+register_condition = specification.register_condition
+# ----------------------------------------------------------------------------
+
+
+@register_condition
+@condition
+def ufo_source(font):
+    import defcon
+    return defcon.Font(font)
+
+
+@register_check
+@check(id='com.google.ufos/check/001', priority=CRITICAL)
```

Comment:
A check id  does not really need to end in a number. We've been doing it that way, but perhaps we could use this as an opportunity to experimenting with doing it differently. Maybe this should end in `/ufolint` instead.

Also, you may prefer to identify the original contribution by using a different choice of reverse domain instead of `com.google`.

I also don't see a point in `.ufos`here. If @davelab6 brings up some ufo-checking need from the Google Fonts project, for detecting some "foo" problem, I would then create a new check in this spec called something like `com.google.fonts/check/foo` and the resulting command line invocation to run this single check would then be:

```
fontbakery check-ufo-sources -c com.google.fonts/check/foo --verbose
``` 

For Dalton Maag contributions for checking some hypothetical "example" issue on UFO sources, then I'd suggest adopting an id such as: `com.daltonmaag/check/example`

**Maybe** it could be `com.domain/check/ufo/something` to make the fact we're dealing with UFOs explicit on the id. But I am not sure about that, yet.

@graphicore do you agree with my considerations here? Or would have other suggestions? This is definitely a work in progress.

Reply:
And **maybe** at some point we could consider adding a command-line attribute like `-d domain` to run only checks from a given domain. In that case, running only the Dalton Maag contributed UFO checks could work like this:

```
fontbakery check-ufo-sources -d com.daltonmaag --verbose
``` 

(possibly allowing multiple -d and -c flags in a single command to cherry-pick what a user wants...)

But I'm just speculating here. Not sure yet what we'll actually do.

Reply:
(How about pytest's `-k` switch:
```
  -k EXPRESSION         only run tests which match the given substring
                        expression. An expression is a python evaluatable
                        expression where all names are substring-matched
                        against test names and their parent classes. Example:
                        -k 'test_method or test_other' matches all test
                        functions and classes whose name contains
                        'test_method' or 'test_other', while -k 'not
                        test_method' matches those that don't contain
                        'test_method' in their names. Additionally keywords
                        are matched to classes and functions containing extra
                        names in their 'extra_keyword_matches' set, as well as
                        functions which have names assigned directly to them.
```
)

Reply:
That looks powerful, but also a bit of over-engineering. I'd probably avoid that complexity.